prov/usnic/src/usdf_fabric.c

```diff
@@ -745,6 +745,9 @@ usdf_getinfo(uint32_t version, const char *node, const char *service,
 	src = NULL;
 	dest = NULL;
 
+	if (!FI_VERSION_GE(fi_version(), version))
+		return -FI_ENODATA;
+
```

Comment:
I'm suggesting that we should insert checks into the framework before it calls the provider getinfo() calls, and see if the providers need any checks beyond that.  My guess is that the providers will not at this point.

Moving forward, I would expect the framework to handle supporting providers with an older version than that used by the framework.